src/cli/commands/beacon.ts

```diff
@@ -1,55 +1,90 @@
 import {CliCommand} from "./interface";
-import * as commander from "commander";
+import {CommanderStatic} from "commander";
+import {parse, JsonMap} from "@iarna/toml";
+import fs from "fs";
 import logger from "../../logger";
-import BeaconNode from "../../node";
-import defaults from "../../node/defaults";
+import BeaconNode, {BeaconNodeCtx} from "../../node";
 import {ethers} from "ethers";
 import {CliError} from "../error";
-import {IApi, IApiConstructor} from "../../rpc/api/interface";
-import {WSServer} from "../../rpc/transport";
+import {IApiConstructor} from "../../rpc/api/interface";
 import * as RPCApis from "../../rpc/api";
-import {JSONRPC} from "../../rpc/protocol";
+import defaults from "../../node/defaults";
+import deepmerge from "deepmerge";
+
+interface IBeaconCommandOptions {
+  db: string;
+  depositContract: string;
+  eth1RpcUrl: string;
+  rpc: string;
+  configFile: string;
+}
+
+interface IConfigFile extends JsonMap{
+  db?: {name: string};
+  chain?: {chain: string};
+  rpc?: {port: string};
+  eth1?: {contract?: {address: string}};
+}
 
 export class BeaconNodeCommand implements CliCommand {
 
-  public register(commander: commander.CommanderStatic): void {
+  public register(commander: CommanderStatic): void {
     commander
       .command("beacon")
       .description("Start lodestar node")
-      .option("-d, --db [db_path]", "Path to file database", defaults.db.name)
-      .option("-c, --depositContract [address]", "Address of deposit contract", defaults.eth1.depositContract.address)
+      .option("-d, --db [db_path]", "Path to file database")
+      .option("-dc, --depositContract [address]", "Address of deposit contract")
       .option("-eth1, --eth1RpcUrl [url]", "Url to eth1 rpc node")
       .option("--rpc [api]", "Exposes the selected RPC api, must be comma separated")
+      .option("-c, --configFile [config_file]", "Config file path")
       .action(async (options) => {
-        //library is not awaiting this method so don't allow error propagation (unhandled promise rejections
+        // library is not awaiting this method so don't allow error propagation
+        // (unhandled promise rejections)
         try {
           await this.action(options);
         } catch (e) {
           logger.error(e.message);
         }
-
       });
   }
 
-  public async action(options: any): Promise<void> {
-    const node = new BeaconNode({
+  public async action(options: IBeaconCommandOptions): Promise<void> {
+    let parsedConfig: IConfigFile;
+    if (options.configFile) {
+      let data: Buffer;
+      try {
+        data = fs.readFileSync(options.configFile);
+      } catch {
+        throw new CliError(`${options.configFile} could not be parsed.`);
+      }
+      parsedConfig = parse(data.toString());
+    }
+    
+    let optionsMap: BeaconNodeCtx = {
       db: {
-        name: options.db
+        name: options.db || parsedConfig ? parsedConfig.db.name : defaults.db.name
```

Comment:
If you apply my deepmerge from comment you should be able to remove this :)

Reply:
will do, I had looked into the library and assumed it wouldn't solve my issue and I thought I had tested it to not fix in my case, and based on your comment I thought you had a very specific use for that config option. I will try it out now to test if it solves my issue, thanks.

Reply:
Yeah, I just tested again, for some reason for specifically the db object (and even when I change type to any) does not merge any values that are in the 
```Typescript
let optionsMap: BeaconNodeCtx = {
      db: {
        name: options.db,
      },
      eth1: {
        contract: {
          address: options.depositContract
        },
        provider: await this.getProvider(options.eth1RpcUrl)
      },
      rpc: {
        apis: this.setupRPC(options.rpc)
      }
};
```
object, but for any values in the toml not matching this pattern are merged. The weirdest part is the eth1.contract.address is merged correctly even though that one is defined as well. I can push a commit to make it easy to test yourself to see. I will leave just the db overriding in just to ensure it still works in the commit. To be clear this issue happens even without the config file flag addition from this commit, the only reason it was working was because a default was passed in through an option and set manually (which can't be used here because of the hierarchy of config needed).

To replicate the issue (When I push my next commit fixing other issues as well):
- remove the logic to handle db name specifically ie: `name: options.db,`
- Run the beacon command with or without config file, you will see that the db will be an empty object when BeaconNode is created